target/linux/rockchip/image/rk3399.mk

```diff
@@ -0,0 +1,15 @@
+# 
+# Copyright (C) 2020 Tobias Maedel
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+define Device/pine64_rockpro64
+	DEVICE_VENDOR := Pine64
+	DEVICE_MODEL := RockPro64
+	DEVICE_DTS := rockchip/rk3399-rockpro64
```

Comment:
Like SUPPORTED_DEVICES, this could also be derived from node name to provide more consistency:

In Device/Default:
```
  DEVICE_DTS_DIR := rockchip
  DEVICE_DTS = $$(SOC)-$(lastword $(subst _, ,$(1)))
```

Here just:
```
SOC := rk3399
```

While this looks more complicated in the first place, it has the advantage that DTS names are "derived" from node names and thus are forced to be consistent. If you still have to deviate, you can still overwrite DEVICE_DTS locally.

Reply:
setting `DEVICE_DTS_DIR` breaks a lot of different things, I think the name is already in use?
anyway: good idea, done!

Reply:
Yes, DEVICE_DTS_DIR does interesting things.
It actually replaces DTS_DIR:
https://github.com/openwrt/openwrt/blob/master/include/image.mk#L525

So, my example was wrong, and you innovative solution should be okay if all (most of the) future devices in this target will be rockchip subfolder.

Another solution would be to have `DTS_DIR := $(DTS_DIR)/rockchip` , but then it would be actually impossible to overwrite that per-device.

All just FYI, I'm okay with how you did it.